src/plugins/data/public/search/types.ts

```diff
@@ -84,3 +98,30 @@ export interface SearchServiceStartDependencies {
   fieldFormats: AggsStartDependencies['fieldFormats'];
   indexPatterns: DataViewsContract;
 }
+
+/**
+ * Format of warnings of failed shards or internal ES timeouts that surface from search responses
+ * @public
+ */
+export interface SearchResponseWarning {
+  /**
+   * type:  for handling the warning in logic
+   */
+  type: 'timed_out' | 'generic_shard_warning' | estypes.ShardFailure['reason']['reason'];
+  /**
+   * isTimeout: true for general internal ES timeout warning
+   */
+  isTimeout?: boolean;
```

Comment:
i don't think we need isTimeout and isShartFailirue, this are already indicated in the type.